scene/gui/control.cpp

```diff
@@ -2792,28 +2792,29 @@ Control::LayoutDirection Control::get_layout_direction() const {
 bool Control::is_layout_rtl() const {
 	if (data.is_rtl_dirty) {
 		const_cast<Control *>(this)->data.is_rtl_dirty = false;
-		if (data.layout_dir == LAYOUT_DIRECTION_INHERITED) {
+		if (GLOBAL_GET(SNAME("internationalization/rendering/force_right_to_left_layout_direction"))) {
+			const_cast<Control *>(this)->data.is_rtl = true;
+		} else if (data.layout_dir == LAYOUT_DIRECTION_INHERITED) {
 			Window *parent_window = get_parent_window();
 			Control *parent_control = get_parent_control();
 			if (parent_control) {
 				const_cast<Control *>(this)->data.is_rtl = parent_control->is_layout_rtl();
 			} else if (parent_window) {
 				const_cast<Control *>(this)->data.is_rtl = parent_window->is_layout_rtl();
 			} else {
```

Comment:
Technically the root node is a Window, right?

So is there a case where we actually reach this `else`? Wouldn't `else if (parent_window)` always be true since there will always be a Window topmost parent, if no Control is in the way?

Reply:
And likewise, how does `internationalization/rendering/root_node_layout_direction` interact with `get_tree().root.layout_direction = LAYOUT_DIRECTION_RTL` for example?

Reply:
`else if (parent_window)` can be false, if the Control's parent is neither a Control nor a Window.
`get_parent_window()` and `get_parent_control()` consider only the direct parent node and not the scene-node-hierarchy.

Reply:
Ah indeed. But then I'm not sure this code is working as intended. If the direct parent is neither Control nor Window, why would it inherit the layout direction from the root node instead of say a Control grandparent or Window grand-grandparent?

Reply:
> If the direct parent is neither Control nor Window

Other nodes do not have defined `width`, and mirroring won't fully work with them. But probably it should check grand-grandparent (as well as Window should check `Control` parents) anyway. Updated.